### PR TITLE
Add aws-serial job for MCO

### PR DESCRIPTION
**Pull request by fabianvf**

### CODE REVIEW

**Comment by deads2k** [comment]
going to make sure the e2e-serial tests at least start running in the rehearse to know we got the right things.

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -348,6 +348,69 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-aws-proxy,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (wking) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:352]: How often has the MCO broken serial tests?  If this is a one-off, I'd consider `always_run: false` to save some CI-volume cost.  This property is one that you can [hand-edit in this file][1]; it is not fed from the config YAML.

I'd also consider `optional: true`, to allow the MCO to merge PRs where serial tests were failing.  But that's mostly to keep their throughput up if serial starts flaking out for some non-MCO reason, and if they only call in the serial tests occasionally (with an `always_run: false` change), requiring serial to pass or be explicitly `/override`n seems reasonable too.

[1]: https://github.com/openshift/ci-tools/blob/7b69d8757ca16b2aec04984a64f34d7f85d68f0c/GENERATOR.md#hand-edited-prow-configuration

Comment (deads2k) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:352]: > How often has the MCO broken serial tests? If this is a one-off, I'd consider `always_run: false` to save some CI-volume cost. This property is one that you can [hand-edit in this file](https://github.com/openshift/ci-tools/blob/7b69d8757ca16b2aec04984a64f34d7f85d68f0c/GENERATOR.md#hand-edited-prow-configuration); it is not fed from the config YAML.

Everyone will say they break it infrequently.  I think it's low enough in the stack to warrant it and there's an explicit scaling test where the machine-api often suspects the MCO.  It came up multiple times in 4.6.

**Comment by deads2k** [comment]
/lgtm

the correct tests ran

**Pull request merged**